### PR TITLE
[YAML] - PubSubLite proto

### PR DESCRIPTION
**Pull request by ffernandez92**
addresses #28685

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by liferoad** [comment]
R: @Polber

**Comment by ffernandez92** [comment]
Run Python_Runners PreCommit 3.9

**Comment by liferoad** [comment]
R: @dpcollins-google

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteReadSchemaTransformProvider.java [hunk 1]
```diff
@@ -214,6 +215,26 @@ public void finish(FinishBundleContext c) {
...
       valueMapper = getRawBytesToRowFunction(beamSchema);
 
+    } else if (format != null && format.equals("PROTO")) {
+      String fileDescriptorPath = configuration.getFileDescriptorPath();
+      String messageName = configuration.getMessageName();
+
+      if (messageName == null) {
+        throw new IllegalArgumentException(
+            "To read from PubSubLite in PROTO format, messageName must be provided.");
+      }
```
Comment (Polber) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteReadSchemaTransformProvider.java:225]: nit: might make sense to have a `validate()` method in the Configuration class that can have checks like this. (and could move the other checks like the one for `inputSchema` below)

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteWriteSchemaTransformProvider.java [hunk 2]
```diff
@@ -211,6 +212,24 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
                 "The input schema must have exactly one field of type byte.");
           }
           toBytesFn = getRowToRawBytesFunction(inputSchema.getField(0).getName());
+        } else if (configuration.getFormat().equals("PROTO")) {
+          String descriptorPath = configuration.getFileDescriptorPath();
+          String schema = configuration.getSchema();
+          String messageName = configuration.getMessageName();
+          if (messageName == null) {
+            throw new IllegalArgumentException("Expecting messageName to be non-null.");
+          }
```
Comment (Polber) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/PubsubLiteWriteSchemaTransformProvider.java:221]: Same here

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/PubsubLiteDlqTest.java [hunk 3]
```diff
@@ -471,4 +534,31 @@ public void testPubsubLiteErrorFnWithDedupingSuccess() {
 
     p.run().waitUntilFinish();
   }
+
+  @Test
+  public void testPubSubLiteErrorFnReadProto() {
```
Comment (Polber) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/PubsubLiteDlqTest.java:539]: I think there should either be an error row that is run through the pipeline to check the error message count, or have a separate failure test case.

File: sdks/java/extensions/protobuf/src/test/java/org/apache/beam/sdk/extensions/protobuf/ProtoByteUtilsTest.java [hunk 4]
```diff
@@ -144,4 +172,23 @@ public void testRowToProtoSchemaFunction() {
...
+            .withFieldValue("address.city", "seattle")
+            .withFieldValue("address.street", "fake street")
+            .withFieldValue("address.zip_code", "TO-1234")
+            .withFieldValue("address.state", "wa")
+            .build();
+
+    Assert.assertNotNull(
+        ProtoByteUtils.getRowToProtoBytesFromSchema(
+                PROTO_STRING_PACKAGE_SCHEMA, "com.test.proto.MyMessage")
+            .apply(row));
```
Comment (Polber) [inline sdks/java/extensions/protobuf/src/test/java/org/apache/beam/sdk/extensions/protobuf/ProtoByteUtilsTest.java:192]: I think having at least 1 test case for `getRowToProtoBytesFromSchema` that validates the output proto after invoking `.apply()` would be beneficial (whether it be this test or one of the existing ones)

**Review by Polber — CHANGES_REQUESTED** [review]
Thanks for the contribution! Looks good overall - I just left a few minor comments.

File: sdks/java/extensions/protobuf/src/main/java/org/apache/beam/sdk/extensions/protobuf/ProtoByteUtils.java
[the remark this answers is not in the document]
Comment (ffernandez92) [reply sdks/java/extensions/protobuf/src/main/java/org/apache/beam/sdk/extensions/protobuf/ProtoByteUtils.java:1]: That was my expectation as well. However, when I tried with the previous implementation (that didn't have the GCS differentiation), the job failed with: `No filesystem found for scheme gs`

**Comment by ffernandez92** [comment]
Run Java_Examples_Dataflow PreCommit 
Run Java_GCP_IO_Direct PreCommit

**Comment by ffernandez92** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by ffernandez92** [comment]
Run Python_Runners PreCommit 3.8

**Comment by ffernandez92** [comment]
Run Python_PVR_Flink PreCommit

**Comment by ffernandez92** [comment]
Run Python_Runners PreCommit 3.8

**Comment by ffernandez92** [comment]
Run Python_Coverage PreCommit

**Comment by ffernandez92** [comment]
Run Python_Integration PreCommit 3.11

**Review by brucearctor — APPROVED** [review]
LGTM

**Pull request merged**